src/main/java/application/rest/v1/actions/AppPodlistFunction.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package application.rest.v1.actions;
+
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import application.rest.v1.ComponentInfoRegistry;
+import application.rest.v1.KAppNavEndpoint;
+import application.rest.v1.Selector;
+import application.rest.v1.ComponentKind;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+
+// apppodlist() or apppodlist(<application-namespace>,<application-name>)
+// Returns a list of pod names from all deployments belonging to specified application.
+// Return value structure is: { "pods" : "[<name1>,<name2>,etc]" }
+public class AppPodlistFunction implements Function {
+    
+    private static final String PODS_PROPERTY_NAME = "pods";
+    private static final String DEPLOYMENT_KIND = "Deployment";
+    private static final String APPLICATION_KIND = "Application";
+    private static final String POD_KIND = "Pod";
+    private static final String METADATA_PROPERTY_NAME = "metadata";
+
+    @Override
+    public String getName() {
+        return "apppodlist";
+    }
+
+    @Override
+    public boolean allowedParameterCount(int parameterCount) {
+        return parameterCount == 0 || parameterCount == 2;
+    }
+
+    @Override
+    public String invoke(ResolutionContext context, List<String> parameters) {
+        
+        final ApiClient client = context.getApiClient();
+        final ComponentInfoRegistry registry = context.getComponentInfoRegistry();
+        final JsonObject resource; 
+        String result = null;     
+
+        if (parameters.size() == 0) {
+            if (!APPLICATION_KIND.equals(context.getResourceKind())) {
+                // The context resource isn't a deployment.
```

Comment:
Fix comment: deployment --> application.
```suggestion
                // The context resource isn't an application.
```